app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardViewModel.kt

```diff
@@ -67,34 +66,15 @@ class DashboardViewModel(
     }
 
     fun observeContactsHealthStatus(): Observable<HealthStatusData> {
-        return Observables.combineLatest(
-            infectionMessengerRepository.observeReceivedInfectionMessages(),
-            quarantineRepository.observeQuarantineState(),
-            configurationRepository.observeConfiguration()
-        ).map { (infectionMessageList, quarantineStatus, configuration) ->
-            val filteredInfectionMessages = infectionMessageList.filter { it.messageType != MessageType.Revoke.Suspicion }
-            Triple(filteredInfectionMessages, quarantineStatus, configuration)
-        }.map { (infectionMessageList, quarantineStatus, configuration) ->
-            if (infectionMessageList.isNotEmpty()) {
-                val redWarningQuarantineThreshold = ZonedDateTime.now().minusHours(
-                    (configuration.redWarningQuarantine ?: DEFAULT_RED_WARNING_QUARANTINE).toLong()
-                )
-                val yellowWarningQuarantineThreshold = ZonedDateTime.now().minusHours(
-                    (configuration.yellowWarningQuarantine ?: DEFAULT_YELLOW_WARNING_QUARANTINE).toLong()
-                )
-                HealthStatusData.ContactsSicknessInfo(
-                    infectionMessageList
-                        .filter { it.timeStamp > redWarningQuarantineThreshold }
-                        .count { it.messageType == MessageType.InfectionLevel.Red },
-                    infectionMessageList
-                        .filter { it.timeStamp > yellowWarningQuarantineThreshold }
-                        .count { it.messageType == MessageType.InfectionLevel.Yellow },
-                    quarantineStatus
-                )
-            } else {
-                HealthStatusData.NoHealthStatus
+        return quarantineRepository.observeQuarantineState()
+            .map { quarantineStatus ->
+                val warningType = quarantineRepository.getCurrentWarningType()
```

Comment:
I would say the warningType should be part of quarantineStatus. Without it it's disconnected and can be used in not sync way.

Reply:
For this method I want to raise also a big red flag, I think we are currently over-simplifying the logic here. I think we should look careful on the old implementation and see how it's connected with the overall app state.

Reply:
@falkorichter , please take note